modules/javafx.graphics/src/main/java/javafx/css/Match.java

```diff
@@ -40,24 +41,26 @@
  */
 public final class Match implements Comparable<Match> {
 
-    final Selector selector;
-    final PseudoClassState pseudoClasses;
-    final int idCount;
+    private final Selector selector;
+    private final Set<PseudoClass> pseudoClasses;
+
     final int styleClassCount;
+    final int idCount;
 
     // CSS3 spec gives weight to id count, then style class count,
     // then pseudoclass count, and finally matching types (i.e., java name count)
     final int specificity;
 
-    Match(final Selector selector, PseudoClassState pseudoClasses, int idCount, int styleClassCount) {
-        assert selector != null;
+    Match(final Selector selector, Set<PseudoClass> pseudoClasses, int idCount, int styleClassCount) {
```

Comment:
just curious: do we really need a 'final' for an effectively final argument?

Reply:
`final` here will just warn you if you try to reassign the parameter.  It was present in the original code.

Reassigning locals and parameters should generally be avoided. For the first, you can have your IDE generate a warning or error.  For the second, I have the habit of always assigning a variable when it is declared.  In the rare cases where this can't be done, the `final` keyword offers little benefit.

The keyword does not offer any further benefits on parameters and locals, and it hasn't since Java added the concept of effectively final (in Java 8 I think).  Before this, you sometimes needed to add `final` so a local could be referred to in anonymous subclasses.